src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/RefDictionary.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Collections.Extensions
+{
+    public sealed class RefDictionary<TKey, TValue> : IEnumerable<(TKey Key, TValue Value)> where TValue : new()
```

Comment:
Why is `TValue` constrained to `new()`? Also why does this implement `IEnumerable<(TKey Key, TValue Value)>` instead of `IEnumerable<KeyValuePair<TKey, TValue>>`?